Add W91 pwm drivers

UR6LAL opened the pull request:
Added PWM drivers implementation for W91 platform

andriy-bilynskyy commented on boards/riscv/tlsr9118bdk40d/tlsr9118bdk40d-common.dtsi:
```diff
@@ -113,3 +125,9 @@
+
+&pwm0 {
+	status = "okay";
+	clock-frequency = <1000000>;
```
Where do we set this frequency?
Is it some constant value? If yes set it in the yaml.

UR6LAL replied:
This value from FREERTOS core, but I've just found that we can use API to get this value. I will rework this part

andriy-bilynskyy replied:
as I see the situation is the same...
set It to const, othewise changing this value will fail the driver.

UR6LAL replied:
I don't use this value anymore, I just forgot to remove it

andriy-bilynskyy commented on drivers/pwm/pwm_w91.c:
```diff
@@ -0,0 +1,202 @@
+/* Driver dts compatibility: telink,w91_pwm */
+#define DT_DRV_COMPAT telink_w91_pwm
+
+#define MAX_CYCLE 1638u
```
Is it value from HW? usually this value is 2^n

UR6LAL replied:
This value isdescribed in "tlsr_timer.h", but there is no define for this:
struct tlsr_timer_cfg_pwm {
	uint16_t high; /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint16_t low;  /* usec, up to ~1638 usec (UINT16_MAX / 40MHz) */
	uint8_t park;  /* level of PWM signal when the timer is stopped */
};

andriy-bilynskyy commented on drivers/pwm/Kconfig.w91:
```diff
@@ -0,0 +1,19 @@
+
+config PWM_TELINK_W91_IPC_RESPONSE_TIMEOUT_MS
+	int "Telink IPC dispatcher timeout while waiting from the endpoint response"
+	default 100
```
Do we really need such value? Most drivers have 10

UR6LAL replied:
I will try to use 10 ms

UR6LAL committed:
drivers: pwm: add W91 pwm drivers

Added W91 pwm drivers implementation

Signed-off-by: Dmytro Kashkarov <ur6lal@gmail.com>

drivers: pwm: add w91 pwm drivers

Added w91 pwm drivers implementation

Signed-off-by: Dmytro Kashkarov <ur6lal@gmail.com>
